portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,20 +30,26 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet { 
 
-  ArrayList<String> cmtarr = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json");
-    String json = new Gson().toJson(cmtarr);
-    response.getWriter().println(json);
+    //String json = new Gson().toJson(cmtarr);
+    //response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     String text = getParameter(request, "comment-input", "");
-    cmtarr.add(text);
+    long timestamp = System.currentTimeMillis();
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("comment-input", text);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Will creating this object only once makes sense then put this into init() and create object only once rather on new on every request.